data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/prepper/ServiceMapStatefulPrepper.java

```diff
@@ -41,17 +51,20 @@
     private volatile static MapDbPrepperState<String> previousTraceGroupWindow;
     private volatile static MapDbPrepperState<String> currentTraceGroupWindow;
     //TODO: Consider keeping this state in lmdb
-    private volatile static  HashSet<ServiceMapRelationship> relationshipState = new HashSet<>();
+    private volatile static HashSet<ServiceMapRelationship> relationshipState = new HashSet<>();
     private static File dbPath;
     private static Clock clock;
     private static int processWorkers;
 
     private final int thisPrepperId;
+    private final Pair<MapDbPrepperState<ServiceMapStateData>, MapDbPrepperState<ServiceMapStateData>> spanDbPair;
+    private final Pair<MapDbPrepperState<String>, MapDbPrepperState<String>> traceDbPair;
```

Comment:
Have you tried creating the pair inside of the call to plugin metrics? I am not sure if the object will get garbage collected, but it would be better to not have to declare these pairs if possible.

Reply:
I will look into this.